compaction/task_manager_module.hh

```diff
@@ -195,18 +195,22 @@ protected:
     virtual future<> run() override = 0;
 };
 
+using do_flush_tables = bool_class<class do_flush_tables_tag>;
```

Comment:
We have already `major_compaction_task_impl::flush_mode`.
I'd rather we generalize it so it can be reused by both cleanup and major compaction rather than defining a new flag.

Reply:
thanks for exposing `flush_mode`
nit: now `do_flush_tables` can be removed